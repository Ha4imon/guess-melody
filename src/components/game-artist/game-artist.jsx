import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {AudioPlayer} from "../audio-player/audio-player.jsx";

class GameArtist extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      isPlayerPlaiyng: false,
    };
  }

  render() {
    const {question, onAnswer} = this.props;

    return (
      <section className="game game--artist">
        <header className="game__header">
          <a className="game__back" href="#">
            <span className="visually-hidden">Сыграть ещё раз</span>
            <img
              className="game__logo"
              src="img/melody-logo-ginger.png"
              alt="Угадай мелодию"
            />
          </a>

          <div className="timer__value" xmlns="http://www.w3.org/1999/xhtml">
            <span className="timer__mins">05</span>
            <span className="timer__dots">:</span>
            <span className="timer__secs">00</span>
          </div>

          <div className="game__mistakes">
            <div className="wrong"></div>
            <div className="wrong"></div>
            <div className="wrong"></div>
          </div>
        </header>

        <section className="game__screen">
          <h2 className="game__title">Кто исполняет эту песню?</h2>
          <div className="game__track">
            <AudioPlayer
              src={question.song.src}
              isPlaying={this.state.isPlayerPlaiyng}
              onButtonClick={() => {
                this.setState((prevState) => ({
                  isPlayerPlaiyng: !prevState.isPlayerPlaiyng,
                }));
              }}
            />
          </div>

          <form
            className="game__artist"
            onChange={(evt) => {
              evt.preventDefault();
              onAnswer();
            }}
          >
            {question.answers.map((item, index) => {
              return (
                <div className="artist" key={item.id}>
                  <input
                    className="artist__input visually-hidden"
                    type="radio"
                    name="answer"
                    value={`artist-${item.type}-${index}`}
                    id={`artist-${item.type}-${index}`}
                  />
                  <label
                    className="artist__name"
                    htmlFor={`artist-${item.type}-${index}`}
                  >
                    <img
                      className="artist__picture"
                      src={item.picture}
                      alt={item.artist}
                    />
                    Пелагея
                  </label>
                </div>
              );
            })}
          </form>
        </section>
      </section>
    );
  }
}

GameArtist.propTypes = {
  onAnswer: PropTypes.func.isRequired,
  question: PropTypes.exact({
    type: PropTypes.oneOf([`artist`]),
    song: PropTypes.exact({
      artist: PropTypes.string.isRequired,
      src: PropTypes.string.isRequired,
    }),
    answers: PropTypes.arrayOf(
        PropTypes.exact({
          id: PropTypes.string.isRequired,
          picture: PropTypes.string.isRequired,
          artist: PropTypes.string.isRequired,
        })
    ),
  }).isRequired,
};

export {GameArtist};
