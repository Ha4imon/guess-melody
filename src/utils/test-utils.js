const createNodeMock = function (element) {
  if (element.type === `audio`) {
    // This is your fake DOM node for <p>.
    // Feel free to add any stub methods, e.g. focus() or any
    // other methods necessary to prevent crashes in your components.
    return {
      src: ``,
    };
  }
  // You can return any object from this method for any type of DOM component.
  // React will use it as a ref instead of a DOM node when snapshot testing.
  return null;
};

export {createNodeMock};
