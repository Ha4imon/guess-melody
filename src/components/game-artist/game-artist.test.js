import React from "react";
import renderer from "react-test-renderer";
import {GameArtist} from "./game-artist";
import {createNodeMock} from "utils/test-utils.js";

it(`Компонент game-artist рендерится`, () => {
  const options = {createNodeMock};
  const question = {
    type: `artist`,
    song: {
      artist: `Jim Snow`,
      src: `audio/sovet-russia.oga`,
    },
    answers: [
      {
        id: `id-artist1`,
        picture: `http://placehold.it/134x134`,
        artist: `John Snow`,
      },
      {
        id: `id-artist2`,
        picture: `http://placehold.it/134x134`,
        artist: `Jack Snow`,
      },
      {
        id: `id-artist3`,
        picture: `http://placehold.it/134x134`,
        artist: `Джек`,
      },
    ],
  };
  const tree = renderer
    .create(<GameArtist question={question} onAnswer={jest.fn()} />, options)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
