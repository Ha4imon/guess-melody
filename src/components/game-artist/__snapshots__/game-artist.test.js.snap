// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент game-artist рендерится 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <button
        className="track__button track__button--play"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
    </div>
    <form
      className="game__artist"
      onChange={[Function]}
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="artist-undefined-0"
          name="answer"
          type="radio"
          value="artist-undefined-0"
        />
        <label
          className="artist__name"
          htmlFor="artist-undefined-0"
        >
          <img
            alt="John Snow"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Пелагея
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="artist-undefined-1"
          name="answer"
          type="radio"
          value="artist-undefined-1"
        />
        <label
          className="artist__name"
          htmlFor="artist-undefined-1"
        >
          <img
            alt="Jack Snow"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Пелагея
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="artist-undefined-2"
          name="answer"
          type="radio"
          value="artist-undefined-2"
        />
        <label
          className="artist__name"
          htmlFor="artist-undefined-2"
        >
          <img
            alt="Джек"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Пелагея
        </label>
      </div>
    </form>
  </section>
</section>
`;
